cmake/fetch_newlib.cmake

```diff
@@ -11,14 +11,21 @@ if(NOT VERSIONS_JSON)
 endif()
 read_repo_version(newlib newlib)
 
-set(newlib_patch ${CMAKE_CURRENT_LIST_DIR}/../patches/newlib.patch)
+set(patch_script ${CMAKE_CURRENT_LIST_DIR}/patch_repo.py)
```

Comment:
There's some code duplication here. Perhaps it could be refactored into (yet) another include file that would define `patch_dir`, and a combined `patch_script_command` list variable containing all of `${Python3_EXECUTABLE} ${patch_script} ${patch_script_args}`?

Then the LLVM file would still be able to have its special case of conditionally adding a second invocation of the whole thing, but there'd be just one place to add further refinements like `GIT_PATCH_METHOD` (which I like!)

Reply:
That would be neater, so I've wrapped it into a simple function which generates the required command for a given target folder, which the LLVM version can simply call twice.

Reply:
I think you've slightly improved on my suggestion, by folding _all_ use of `patch_dir` into the include file instead of having it just defined there and used by the caller. I like it!